client-core/src/wallet/syncer.rs

```diff
@@ -374,6 +381,52 @@ impl<'a, S: SecureStorage, C: Client, D: TxDecryptor> WalletSyncerImpl<'a, S, C,
         Ok(())
     }
 
+    fn sync_tx(&mut self, tx_str: &str) -> Result<Coin> {
+        let tx_raw = base64::decode(tx_str)
+            .chain(|| (ErrorKind::DecryptionError, "Unable to decrypt transaction"))?;
+        let tx_info: TransactionInfo = serde_json::from_slice(&tx_raw)
+            .chain(|| (ErrorKind::DecryptionError, "Unable to decrypt transaction"))?;
+        // check if the output is spent or not
+        let v = self
+            .env
+            .client
+            .query("meta", &tx_info.tx.id().to_vec())?
+            .bytes()?;
+        let bit_flag = BitVec::from_bytes(&v);
+        let spent_flags: Result<Vec<bool>> = tx_info
+            .tx
+            .outputs()
+            .iter()
+            .enumerate()
+            .map(|(index, _output)| {
+                bit_flag
+                    .get(index)
+                    .chain(|| (ErrorKind::InvalidInput, "check failed in enclave"))
+            })
+            .collect();
+        let mut memento = WalletStateMemento::default();
+        // check if tx belongs to the block
+        let block = self.env.client.block(tx_info.block_height)?;
+        if !block.enclave_transaction_ids()?.contains(&tx_info.tx.id()) {
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                "block height and transaction not match",
+            ));
+        }
+        let imported_vlaue = import_transaction(
+            &self.wallet,
+            &self.wallet_state,
+            &mut memento,
+            &tx_info.tx,
+            tx_info.block_height,
+            block.header.time,
+            spent_flags?,
+        )
+        .chain(|| (ErrorKind::InvalidInput, "import error"))?;
+        self.save(&memento)?;
```

Comment:
This is the side-effect of implement this feature in `Syncer`, `save` will also write `SyncState` which is not used by this feature, but if there is concurrently running syncing process, it will override data.

Reply:
I'll make an individual function called `update_status` and use it in `save`